txcache/selectionTracker.go

```diff
@@ -110,29 +110,27 @@ func (st *selectionTracker) validateTrackedBlocks(chainOfTrackedBlocks []*tracke
 
 	for _, tb := range chainOfTrackedBlocks {
 		for address, breadcrumb := range tb.breadcrumbsByAddress {
-			// TODO make sure that the accounts which don't yet exist are properly handled
-			accountState, err := session.GetAccountState([]byte(address))
+			initialNonce, initialBalance, _, err := session.GetAccountNonceAndBalance([]byte(address))
 			if err != nil {
-				log.Debug("selectionTracker.validateTrackedBlocks",
-					"err", err)
+				// TODO: Maybe add more context to this error?
+				log.Debug("selectionTracker.validateTrackedBlocks", "err", err)
 				return err
 			}
 
-			// validate that a breadcrumb is continuous
-			if !validator.continuousBreadcrumb(address, breadcrumb, accountState) {
+			if !validator.continuousBreadcrumb(address, initialNonce, breadcrumb) {
```

Comment:
@danielradu10, maybe another TODO: check if we can improve the consistency for the order of parameters. E.g. sometimes we have _nonce, breadcrumb_, and sometimes _breadcrumb, nonce_.